File: mojang/lib/client/java.ts
import { z } from 'zod';

export interface JavaVersion {
  readonly component: string;
  readonly majorVersion: number;
}

export const JavaVersionSchema = z
  .object({ component: z.string(), majorVersion: z.number() })
  .default({ component: 'jre-legacy', majorVersion: 8 });

File: crates/opys-mojang/src/java.rs
use serde::{Deserialize, Serialize};

/// Missing entirely on 1.6.x and its snapshots, hence the default.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JavaVersion {
    pub component: String,
    pub major_version: u32,
}

impl Default for JavaVersion {
    fn default() -> Self {
        Self {
            component: "jre-legacy".to_owned(),
            major_version: 8,
        }
    }
}
